components/new_post/post_component.tsx

```diff
@@ -0,0 +1,500 @@
+                    id='search_item.thread'
+                    defaultMessage='Thread in {channel}'
+                    values={{
+                        channel: channelName,
+                    }}
+                />
+            );
+        } else if (isDirectMessage) {
+            name = (
+                <FormattedMessage
+                    id='search_item.direct'
+                    defaultMessage='Direct Message (with {username})'
+                    values={{
+                        username: props.displayName,
+                    }}
+                />
+            );
+        }
+
+        return name;
+    };
+
+    const getClassName = () => {
+        const post = props.post;
+        const isSystemMessage = PostUtils.isSystemMessage(post);
+        const isMeMessage = checkIsMeMessage(post);
+        const hovered =
+            fileDropdownOpened || dropdownOpened || a11yActive || isPostBeingEdited;
+        return classNames('a11y__section post', {
+            'post--highlight': props.shouldHighlight && !fadeOutHighlight,
+            'same--root': hasSameRoot(props),
+            'post--bot': PostUtils.isFromBot(post),
+            'post--pinned-or-flagged-highlight': props.shouldHighlight && (post.is_pinned || props.isFlagged),
+            'post--editing': props.isPostBeingEdited,
+            'current--user': props.currentUserId === post.user_id,
+            'post--system': isSystemMessage || isMeMessage,
+            'post--root': props.hasReplies && !(post.root_id && post.root_id.length > 0),
+            'post--comment': post.root_id && post.root_id.length > 0,
+            'post--compact': props.compactDisplay,
+            'post--hovered': hovered,
+            'same--user': props.isConsecutivePost,
+            'cursor--pointer': alt && !props.channelIsArchived,
+            'post--hide-controls': post.failed || post.state === Posts.POST_DELETED,
+            'post--comment same--root': PostUtils.fromAutoResponder(post),
+            'post--pinned-or-flagged': post.is_pinned || props.isFlagged,
+        });
+    };
+
+    const handleAlt = (e: KeyboardEvent) => {
+        if (alt !== e.altKey) {
+            setAlt(e.altKey);
+        }
+    };
+
+    const handleFileDropdownOpened = useCallback((open: boolean) => setFileDropdownOpened(open), []);
+
+    const handleDropdownOpened = useCallback((opened: boolean) => {
+        if (props.togglePostMenu) {
+            props.togglePostMenu(opened);
+        }
+        setDropdownOpened(opened);
+    }, []);
+
+    const handleMouseOver = (e: MouseEvent<HTMLDivElement>) => {
+        setHover(true);
+        setAlt(e.altKey);
+    };
+
+    const handleMouseLeave = () => {
+        setHover(false);
+        setAlt(false);
+    };
+
+    const addKeyboardListeners = () => {
+        document.addEventListener('keydown', handleAlt);
+        document.addEventListener('keyup', handleAlt);
+    };
+
+    const removeKeyboardListeners = () => {
```

Comment:
are we removing this listeners on unmount somewhere?

Reply:
AFAIK with useEffect you don't think of the component lifecycle, but rather manipulate based on prop change. So I have split the two useEffects by concern. But I am open to a suggestion on where to add the removal of the listeners too?